arangod/RocksDBEngine/RocksDBReplicationContext.cpp

```diff
@@ -797,17 +797,33 @@ RocksDBReplicationContext::CollectionIterator::CollectionIterator(
       _isUsed{false},
       _sortedIterator{!sorted} // this makes sure that setSorted works
 {
-  TRI_ASSERT(snapshot != nullptr);
+  TRI_ASSERT(snapshot != nullptr && coll);
   _readOptions.snapshot = snapshot;
   _readOptions.verify_checksums = false;
   _readOptions.fill_cache = false;
   _readOptions.prefix_same_as_start = true;
   
+  if (!vocbase.use()) { // false if vobase was deleted
+    THROW_ARANGO_EXCEPTION(TRI_ERROR_ARANGO_DATABASE_NOT_FOUND);
+  }
+  TRI_vocbase_col_status_e ignore;
+  int res = vocbase.useCollection(logical.get(), ignore);
+  if (res != TRI_ERROR_NO_ERROR) { // collection was deleted
+    THROW_ARANGO_EXCEPTION(TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND);
```

Comment:
Does this have to _unuse_ the database again, or is that ensured by some other means?